### PR TITLE
feat: DIA-868: Component library + Chip Input

### PR DESCRIPTION
**Pull request by nick-skriabin**
Introducing the UI component library with its first citizen - the ChipInput component.

UI library contains:
- UI (React) component written in TypeScript and styled with SCSS
- Assets library -- icons in SVG format
- Components tests -- Cypress
- Storybook with interactive access to the components and/or components composition

This work was initiated by DIA-868 to support so-called "classes input" (see designs in the ticket). The basis for this component is "MultiEmailInput" from LSE's invitation flow. The component was refactored from JS+Stylus into TS+SCSS, and generalized to be universal.

The work on the component library was initiated due to inability to test components that are not the part of the UI yet.

Note: Name "Classes component" comes from the upcoming PW feature, but the general name for the underlying component is "ChipInput".

### PR fulfills these requirements
- [x] Tests for the changes have been added/updated
- [ ] Docs have been added/updated
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance

### What feature flags were used to cover this change?
None

### What alternative approaches were there?
Implement the whole interface and include a bunch of tickets in a single PR.

### This change affects (describe how if yes)
- [ ] Performance
- [ ] Security
- [x] UX - new generalized chip component to support future work
- [x] DX - introduction of the component library must improve DX by avoiding code duplication, scope creep. It also improves code quality as every component must be covered with tests.

### Does this PR introduce a breaking change?
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
- [ ] e2e (codecept)
- [x] integration (cypress)
- [ ] unit (jest)

### Which logical domain(s) does this change affect?
Monorepo, PW

### CODE REVIEW

File: web/libs/storybook/src/components/ChipInput.scss [hunk 1]
```diff
@@ -0,0 +1,96 @@
+@import "../assets/styles/base.scss";
+
+$danger_color: #ff0000;
```
Comment (yyassi-heartex) [inline web/libs/storybook/src/components/ChipInput.scss]: might be a prudent to add the danger color to base.scss to make it available to other components as more components get added here

File: web/libs/storybook/tests/support/e2e.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { CURRENT_FLAGS } from '../feature-flags';
+import '@humansignal/frontend-test/cypress/support/e2e'
+
+Cypress.on('uncaught:exception', (err, runnable) => {
```
Comment (hlomzik) [inline web/libs/storybook/tests/support/e2e.ts]:
```suggestion
Cypress.on('uncaught:exception', () => {
```

File: web/libs/storybook/tests/support/e2e.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  //it is to prevent outliner failing tests with uncaught error
+  return false;
+});
+
+beforeEach(() => {
+  cy.on('uncaught:exception', err => {
+    return !err.message.includes('ResizeObserver loop completed with undelivered notifications.');
```
Comment (hlomzik) [inline web/libs/storybook/tests/support/e2e.ts]: Most probably we have some other errors, I remember @Gondragos added similar ignore procedure for them. And potentially we could have such error, so better to use array of errors to ignore.

File: web/libs/storybook/.storybook/main.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    options: {},
+  },
+
+  viteFinal: async (config) =>
+    mergeConfig(config, {
+      plugins: [
+        nxViteTsPaths(),
+        svgr({
+          esbuildOptions: {},
+          svgrOptions: { plugins: ['@svgr/plugin-jsx'], ref: true },
```
Comment (hlomzik) [inline web/libs/storybook/.storybook/main.ts:24]: Not sure how does this work exactly, but in all our other configs we also use `removeViewBox: false` for proper work of some icons.

File: web/libs/storybook/.storybook/preview-head.html [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    --gray-8: #9d9d9d;
+    --gray-9: #8c8c8c;
+
+    --theme-error: var(--red-6);
+    --theme-text: #171819;
+    --theme-white: white;
+    --theme-accent: var(--blue-6);
+    --theme-bg: white; /*var(--gray-2)*/
+
+    --pirmary-on-containte: #052861;
```
Comment (hlomzik) [inline web/libs/storybook/.storybook/preview-head.html]: Definitely typo in "primary" and I'm not sure I understand the last word

Comment (hlomzik) [reply web/libs/storybook/.storybook/preview-head.html]: and in other places as well

File: web/libs/storybook/README.md
Comment (hlomzik) [inline web/libs/storybook/README.md:1]: Would be great to use this README to describe main principles on how to work with storybook here. For example should we really put components directly here, which ones and why?

**Review by hlomzik — CHANGES_REQUESTED** [review]

File: web/libs/storybook/tests/support/e2e.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    console.log('Setting feature flags', CURRENT_FLAGS);
+    Object.assign(win, {
+      DISABLE_DEFAULT_LSF_INIT: true,
+      APP_SETTINGS: {
+        ...(win.APP_SETTINGS ?? {}),
+        feature_flags: CURRENT_FLAGS,
+      },
+    });
+  });
+  cy.log('Feature flags set');
```
Comment (hlomzik) [inline web/libs/storybook/tests/support/e2e.ts]: we have both cy.log and console.log here. do we need them?

File: web/libs/storybook/.storybook/main.ts (hunk 4 above)
Comment (nick-skriabin) [reply web/libs/storybook/.storybook/main.ts:24]: just checked, it doesn't remove it, so we're good

File: web/libs/storybook/tests/support/e2e.ts (hunk 3 above)
Comment (nick-skriabin) [reply web/libs/storybook/tests/support/e2e.ts]: ty, reorganized the setup here. I also removed whole e2e section as it's the same thing in cypress. so we only have components tests there now

File: web/libs/storybook/README.md
Comment (nick-skriabin) [reply web/libs/storybook/README.md:1]: still working on it. there'l be readme + separate doc about the components lib

File: web/libs/storybook/tests/support/e2e.ts (hunk 6 above)
Comment (nick-skriabin) [reply web/libs/storybook/tests/support/e2e.ts]: removed it for now completely as we don't have component-level feature flags yet

File: web/libs/storybook/src/components/ChipInput.scss (hunk 1 above)
Comment (nick-skriabin) [reply web/libs/storybook/src/components/ChipInput.scss]: replaced with an existing css var

File: web/libs/storybook/.storybook/main.ts (hunk 4 above)
Comment (hlomzik) [reply web/libs/storybook/.storybook/main.ts:24]: ah, yes, NX does this automatically, that's why it was good in monorepo and failed during back-porting!

File: web/libs/storybook/.storybook/preview-head.html (hunk 5 above)
Comment (hlomzik) [reply web/libs/storybook/.storybook/preview-head.html]: other places fixed but this exact line is still untouched :D

**Review by hlomzik — APPROVED** [review]

File: web/libs/storybook/src/assets/icons/index.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+/// <reference types="vite-plugin-svgr/client" />
+
```
Comment (yyassi-heartex) [inline web/libs/storybook/src/assets/icons/index.ts:2]: this is a good opportunity to clean things up
```suggestion
```

Comment (nick-skriabin) [reply web/libs/storybook/src/assets/icons/index.ts:2]: without it for some reason TS complains about types (can't import things with `?react`). gotta figure out a different way. but this is recommended by the svgr themself

Comment (yyassi-heartex) [reply web/libs/storybook/src/assets/icons/index.ts:2]: interesting, then lets disregard that - thanks for following up!

**Review by yyassi-heartex — APPROVED** [review]

**Review by MihajloHoma — APPROVED** [review]

**Pull request merged**